### PR TITLE
Tweaks of disabling runtime environments

### PR DESCRIPTION
**Pull request by Teyras**

### CODE REVIEW

File: app/V1Module/presenters/AssignmentsPresenter.php [hunk 1]
```diff
@@ -193,9 +193,12 @@ public function actionUpdateDetail(string $id) {
     }
 
     if ($this->isRequestJson()) {
-      $disabledRuntimeIds = $req->getPost("disabledRuntimeEnvironments");
-      $disabledRuntimes = array_map([$this->runtimeEnvironments, "findOrThrow"], $disabledRuntimeIds);
+      $disabledRuntimeIds = $req->getPost("disabledRuntimeEnvironmentIds");
```
Comment (Neloop) [inline app/V1Module/presenters/AssignmentsPresenter.php]: What if `$disableRuntimeIds` is null?

File: app/model/entity/Assignment.php [hunk 2]
```diff
@@ -357,7 +357,9 @@ function ($key, RuntimeEnvironment $env) {
       "maxPointsBeforeFirstDeadline" => $this->maxPointsBeforeFirstDeadline,
       "maxPointsBeforeSecondDeadline" => $this->maxPointsBeforeSecondDeadline,
       "submissionsCountLimit" => $this->submissionsCountLimit,
-      "runtimeEnvironmentsIds" => $this->getRuntimeEnvironmentsIds(),
```
Comment (Neloop) [inline app/model/entity/Assignment.php]: just a thought, what about adding new item (`exerciseRuntimeEnvironmentIds`) and keep the old one?

File: app/V1Module/presenters/AssignmentsPresenter.php [hunk 3]
```diff
@@ -193,10 +193,15 @@ public function actionUpdateDetail(string $id) {
...
+      if ($disabledRuntimeIds !== null) {
+        if (!is_array($disabledRuntimeIds)) {
```
Comment (Neloop) [inline app/V1Module/presenters/AssignmentsPresenter.php]: Instead of this you can use validation check in `@Param` annotation

Comment (Teyras) [reply app/V1Module/presenters/AssignmentsPresenter.php]: I'm not sure which one do you mean. Null or array validation?

Comment (Neloop) [reply app/V1Module/presenters/AssignmentsPresenter.php]: I'm refering to the line with array check... so I would say array validation

Comment (Teyras) [reply app/V1Module/presenters/AssignmentsPresenter.php]: Yeah, that could be done... I wasn't sure how validation works with optional parameters

**Pull request merged**